config/config_test.go

```diff
@@ -6,14 +6,55 @@ import (
 
 func TestLoadConfig(t *testing.T) {
 
-	config, err := LoadConfigFromFile("test_resources/test1.yaml")
+	config, err := LoadConfigFromFile("test_resources/test_full_conf.yaml")
 	if err != nil {
 		t.Fatalf("Failed to load config: %v", err)
 	}
 
-	t.Logf("Config: %#v", config)
+	for _, pf := range config.Profiles {
+		t.Logf("Profile: %s", pf.ProfileName)
+		for _, pb := range pf.PipelineBlocks {
+			t.Logf("Operation: %s", pb.Operation)
+			if pb.Operation == "resize" {
```

Comment:
Switch! Switch! Switch! :D

```go
switch block.Operation {
	case "resize":
		t.Logf("Resize: %v", block.Resize)
	case "write":
		t.Logf("Write: %v", block.Write)
	case "icc_embed":
		t.Logf("ICC: %v", block.ICCEmbedProfile)
	case "encode":
		t.Logf("Encode: %v", block.Encode)
	case "crop":
		t.Logf("Crop: %v", block.Crop)
	}
}
```

Reply:
This might be a trivial way of using `switch`.
Just like how we use `switch` in C/C++. But without `break` after each case. : D